Pod/Classes/OriginateHTTPClient.m

```diff
@@ -440,51 +306,73 @@ + (void)DELETEResource:(NSString *)URI
                                           NSURLResponse *response,
                                           NSError *responseError)
             {
-                if (!responseBlock) {
-                    return;
-                }
-                
-                NSHTTPURLResponse *HTTPResponse = (NSHTTPURLResponse *)response;
-                NSError *error = [[self class] errorForResponse:HTTPResponse connectionError:responseError];
-                
-                if (error) {
-                    responseBlock(nil, error);
-                    return;
-                }
-                
-                if (data.length == 0) {
-                    if ([[self class] emptyBodyAcceptableForHTTPResponse:HTTPResponse]) {
-                        responseBlock(nil, nil);
-                    }
-                    else {
-                        responseBlock(nil, [self errorEmptyResponse]);
-                    }
-                    return;
-                }
-                
-                NSError *jsonError;
-                NSDictionary *responseBody = [NSJSONSerialization JSONObjectWithData:data
-                                                                             options:NSJSONReadingAllowFragments
-                                                                               error:&jsonError];
-                
-                if (jsonError) {
-                    responseBlock(nil, [[self class] HTTPError500InternalServer]);
-                    return;
-                }
-                
-                OriginateHTTPLog* log = [[OriginateHTTPLog alloc] initWithURLResponse:response
-                                                                       responseObject:responseBody];
-                [[NSNotificationCenter defaultCenter] postNotificationName:OriginateHTTPClientResponseNotification
-                                                                    object:log];
-                
-                responseBlock(responseBody, nil);
+                [[self class] handleResponse:response
+                                        data:data
+                                       error:responseError
+                      evaluateLocationHeader:NO
+                                  completion:responseBlock];
             }];
     
     [task resume];
 }
 
-
-#pragma mark - Empty HTTP body
++ (void)handleResponse:(NSURLResponse *)response
+                  data:(NSData *)data
+                 error:(NSError *)responseError
+evaluateLocationHeader:(BOOL)evaluateLocationHeader
+            completion:(OriginateHTTPClientResponse)responseBlock
+{
+    NSHTTPURLResponse *HTTPResponse = (NSHTTPURLResponse *)response;
```

Comment:
@allewun Take a look at this section specifically to make sure I covered all the bases